[soonami69] iP

soonami69 opened the pull request:
# Clonky
> "I think this is very good" - Goku ([Source](Imadeitup.com))

***Clonky*** helps you store tasks and keep track of what to do.

***Clonky*** is:
- 100% Safe
- Fast
- Works (sometimes, most of the time)

All you need to do is:
1. Download it [here](https://github.com/soonami69/ip/releases/tag/A-Jar)
2. Double-click it
3. Figure the rest out yourself 🦄

Features:
- [x] Manage tasks
- [x] Search for tasks by descriptions
- [ ] Reminders
- [ ] ~~Mine bitcoin~~ Play ASMR

If you are a `Java` programmer, you can run it too<sup>*</sup>. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(Clonky.class, args);
    }
}
```
<sub>*I didn't actually test this</sub>

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

soonami69 committed:
Level 0

soonami69 committed:
Level 1

soonami69 committed:
Level 2

soonami69 committed:
Level 3

soonami69 committed:
Level 4

soonami69 committed:
Janky A-TextUITesting

soonami69 committed:
Error handling

soonami69 committed:
Level 6

soonami69 committed:
Level-7

Refactored the tasks into one package, and added the ability
to load and save tasks in plaintext.

soonami69 committed:
Level 7

Added the ability to save tasks and load them when program is opened.

soonami69 committed:
Merge branch 'branch-Level-7'

soonami69 committed:
Change Deadline and Event to use Java LocalDate instead of String

soonami69 committed:
Merge branch 'branch-Level-8'

soonami69 committed:
Create UI package and made Clonky non-static

soonami69 committed:
Merge branch 'branch-package-moreOOP'

soonami69 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

soonami69 committed:
Add tests for UI and Parser Classes

soonami69 committed:
Add back src/main after erroneously putting it in .gitignore

soonami69 committed:
Add JavaDoc comments to public classes

soonami69 committed:
Reformat all classes to adhere to coding standard

soonami69 committed:
Add TaskList class for Level-9 increment

soonami69 committed:
Merge remote-tracking branch 'origin/branch-A-JavaDoc'

soonami69 committed:
Add JavaFX GUI, and rewrote JUnit tests to fit refactored code

soonami69 committed:
Add checkstyle.xml and reformat code to match

soonami69 committed:
Fix crash when bye command called due to NullPointerException

subiloble commented on src/main/java/clonky/tasks/Event.java:
```diff
@@ -3,16 +3,33 @@
 import java.time.LocalDate;
 import java.time.format.DateTimeFormatter;
 
+/**
+ * The {@code Event} class represents a task that occurs within a specific time frame.
+ * It extends the {@code Task} class and stores {@code LocalDate} values for the start and end dates.
+ */
```
I like your JavaDoc comments which is clear and specific.

subiloble commented on src/main/java/clonky/tasks/Parser.java:
```diff
@@ -23,6 +34,13 @@ public void addTodo(String description) throws NoDescriptionException {
         printTaskCount();
     }
 
+    /**
+     * Adds a new {@code Deadline} task with a due date.
+     *
+     * @param arguments The task description and due date in the format "description /by YYYY-MM-DD".
+     * @throws NoDescriptionException If the description is missing.
+     * @throws NoByException If the due date is missing.
+     */
```
LGTM.

UltimateHG commented on src/main/java/clonky/tasks/Event.java:
```diff
@@ -0,0 +1,39 @@
+    private final LocalDate from;
+    private final LocalDate to;
+
+    /**
+     * Constructs a new {@code Event} task with a description, start date, and end date.
+     *
+     * @param description The description of the event.
+     * @param from The start date of the event in {@code YYYY-MM-DD} format.
+     * @param to The end date of the event in {@code YYYY-MM-DD} format.
+     */
```
I like the style of documentation, clear and concise

UltimateHG commented on src/main/java/clonky/tasks/TaskList.java:
```diff
@@ -0,0 +1,270 @@
+     * @param filePath The file path to save to.
+     * @return A response detailing if the operation was successful or not.
+     */
+    public Response saveTasks(String filePath) {
+        String text = "";
+        Mood mood;
+        Color color;
+        String path = (filePath == null || filePath.isEmpty())
+                ? Paths.get("clonky", "tasks.txt").toString()
+                : filePath;
```
Would it be better to set a user-specifiable save path (i.e. in a config file) instead of accepting an arbitrary filepath as a fully user-controllable variable?

UltimateHG commented on src/main/java/clonky/tasks/TaskList.java:
```diff
@@ -0,0 +1,270 @@
+     * @return A list of tasks that match the given description.
+     */
+    public List<Task> findTasksByDescription(String description) {
+        return tasks.stream()
+                .filter(task -> task.description.toLowerCase().contains(description.toLowerCase()))
+                .collect(Collectors.toList());
```
I like the use of functioning programming here to handle the finding of tasks 👍

soonami69 replied:
Thanks for the suggestion! That's a great idea actually

UltimateHG reviewed (commented):
Clear and well-documented code, looks good.

soonami69 committed:
Add asserts

Asserts were added to ensure, mainly to constructors, to verify that null is not being passed into the constructor.
This helps ensure that all classes are being instantiated properly and to validate expected behavior.

soonami69 committed:
Add assertions to Parser class

soonami69 committed:
Add assertions to Parser class

soonami69 committed:
Add priority level to tasks

soonami69 committed:
Reformat TaskList to adhere to coding standard

soonami69 committed:
Fix typos

soonami69 committed:
Create branch for tracking by dashboard

soonami69 committed:
Add user guide

soonami69 committed:
Edit gradle dependencies to support building on Mac

soonami69 committed:
Fix typos in user guide

soonami69 committed:
Remove UI.png

soonami69 committed:
Add Ui.png

soonami69 committed:
Fix typo in user guide

soonami69 committed:
Update README.md
